GANDLF/GAN/compute/inference_loop.py

```diff
@@ -0,0 +1,206 @@
+    print("Number of dims     : ", parameters["model"]["dimension"])
+    if "num_channels" in parameters["model"]:
+        print("Number of channels : ", parameters["model"]["num_channels"])
+    # ensure outputs are saved properly
+    parameters["save_output"] = True
+
+    assert (
+        parameters["model"]["type"].lower() == "torch"
+    ), f"The model type is not recognized: {parameters['model']['type']}"
+    assert (
+        "save_format" in parameters["inference_config"]
+    ), "The save format is not provided"
+    assert (
+        "n_generated_samples" in parameters["inference_config"]
+    ), "The number of samples to generate is not provided"
+    assert (
+        "batch_size" in parameters["inference_config"]
+    ), "The batch size for inference is not provided"
+    assert parameters["inference_config"]["save_format"] in [
+        "png",
+        "jpeg",
+        "jpg",
+        "nii.gz",
+    ], f"The save format is not recognized: {parameters['inference_config']['save_format']}"
+    if dataframe is not None:
+        warn(
+            "The dataframe was passed, but it's usage is not yet implemented in inference. Ignoring it."
+        )
+
+    assert (
+        parameters["model"]["type"].lower() != "openvino"
+    ), "OpenVINO not yet implemented"
+
+    pytorch_objects = create_pytorch_objects_gan(parameters, device=device)
+    model, parameters = pytorch_objects[0], pytorch_objects[-1]
+    main_dict = None
+
+    if parameters["model"]["type"].lower() == "torch":
+        # Loading the weights into the model
+        if os.path.isdir(modelDir):
+            files_to_check = [
+                os.path.join(
+                    modelDir,
+                    str(parameters["model"]["architecture"])
+                    + best_model_path_end,
+                ),
+                os.path.join(
+                    modelDir,
+                    str(parameters["model"]["architecture"])
+                    + latest_model_path_end,
+                ),
+            ]
+
+            file_to_load = None
+            for best_file in files_to_check:
+                if os.path.isfile(best_file):
+                    file_to_load = best_file
+                    break
+
+            assert file_to_load != None, "The 'best_file' was not found"
+
+        main_dict = torch.load(file_to_load, map_location=parameters["device"])
+        model.load_state_dict(main_dict["model_state_dict"])
+        parameters["previous_parameters"] = main_dict.get("parameters", None)
+        model.eval()
+
+    n_generated_samples = parameters["inference_config"]["n_generated_samples"]
+    latent_vector_size = parameters["model"]["latent_vector_size"]
+    batch_size = parameters["inference_config"]["batch_size"]
+    n_iterations = (
+        n_generated_samples // batch_size
+    )  # how many iterations to run
+    remaining_samples = (
+        n_generated_samples % batch_size
+    )  # remaining samples for last iteration
```

Comment:
Run `black` on this again.
```suggestion
    # how many iterations to run
    n_iterations = (
        n_generated_samples // batch_size
    )  
    # remaining samples for last iteration
    remaining_samples = (
        n_generated_samples % batch_size
    )  
```